spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/MqttPahoMessageDrivenChannelAdapter.java

```diff
@@ -174,6 +183,16 @@ public void setApplicationEventPublisher(ApplicationEventPublisher applicationEv
 		this.applicationEventPublisher = applicationEventPublisher; // NOSONAR (inconsistent synchronization)
 	}
 
+	@Override
+	public MqttConnectOptions getConnectionInfo() {
+		MqttConnectOptions options = this.clientFactory.getConnectionOptions();
+		if (options.getServerURIs() == null && getUrl() != null) {
+			options = MqttUtils.cloneConnectOptions(options);
+			options.setServerURIs(new String[] { getUrl() });
```

Comment:
Better to extract local variable for `URL` to avoid noise from Sonar that we defer NPE calling the method several times.